hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/distributed/NodeQueueLoadMonitor.java

```diff
@@ -261,12 +391,15 @@ protected void onNewNodeAdded(
 
     if (rmNode.getState() != NodeState.DECOMMISSIONING &&
         (estimatedQueueWaitTime != -1 ||
-            comparator == LoadComparator.QUEUE_LENGTH)) {
+            comparator == LoadComparator.QUEUE_LENGTH ||
+            comparator == LoadComparator.QUEUE_LENGTH_THEN_RESOURCES)) {
       this.clusterNodes.put(rmNode.getNodeID(),
```

Comment:
Every second we are taking too many write locks of ClusterNode. IIUC that not the best optimized option. We should probably take the write lock once and update all the values like waitTime,SetQueueLength,labels etc..

Reply:
We could update this in followup PR also..